src/Mvc/Mvc.Core/src/ModelBinding/ParameterBinder.cs

```diff
@@ -82,13 +82,34 @@ public class ParameterBinder
         /// <param name="metadata">The <see cref="ModelMetadata"/>.</param>
         /// <param name="value">The initial model value.</param>
         /// <returns>The result of model binding.</returns>
-        public virtual async Task<ModelBindingResult> BindModelAsync(
+        public virtual Task<ModelBindingResult> BindModelAsync(
             ActionContext actionContext,
             IModelBinder modelBinder,
             IValueProvider valueProvider,
             ParameterDescriptor parameter,
             ModelMetadata metadata,
             object value)
+            => BindModelAsync(actionContext, modelBinder, valueProvider, parameter, metadata, value, container: null).AsTask();
+
+        /// <summary>
+        /// Binds a model specified by <paramref name="parameter"/> using <paramref name="value"/> as the initial value.
+        /// </summary>
+        /// <param name="actionContext">The <see cref="ActionContext"/>.</param>
+        /// <param name="modelBinder">The <see cref="IModelBinder"/>.</param>
+        /// <param name="valueProvider">The <see cref="IValueProvider"/>.</param>
+        /// <param name="parameter">The <see cref="ParameterDescriptor"/></param>
+        /// <param name="metadata">The <see cref="ModelMetadata"/>.</param>
+        /// <param name="value">The initial model value.</param>
+        /// <param name="container">The container for the model.</param>
+        /// <returns>The result of model binding.</returns>
+        public virtual async ValueTask<ModelBindingResult> BindModelAsync(
```

Comment:
What does `ValueTask` buy us in this context?

Reply:
It works fairly well for this scenario

a) We read the model bound value once.
b) In a large number of cases the operation finishes synchronously.

We'd avoided using ValueTask in our APIs previously just because it didn't seem prevalent in the runtime and we didn't want to use something that users might have been unfamiliar with. But now that appears so frequently in the rest of the BCL, this seems a good win. I'll run this by the API review too in case I'm missing some subtlety about it's use that would make it unadvisable here.